File: microservices/notification-service/src/main/java/com/karate/notification_service/infrastructure/feign/EnrollmentClient.java
package com.karate.notification_service.infrastructure.feign;

import com.karate.notification_service.infrastructure.feign.config.FeignClientConfig;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

import java.util.List;

@FeignClient(name = "enrollment-service", configuration = FeignClientConfig.class)
public interface EnrollmentClient {

    @GetMapping("/internal/enrollments/training/{trainingId}/emails")
    List<String> getEnrolledEmails(@PathVariable("trainingId") Long trainingId);
}

File: microservices/feedback-service/src/main/kotlin/com/karate/feedback_service/infrastructure/client/EnrollmentClient.kt
package com.karate.feedback_service.infrastructure.client;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@FeignClient(name = "enrollment-service")
interface EnrollmentClient {
    @GetMapping("/internal/enrollments/enrolled/{userId}/{sessionId}")
    fun checkUserEnrolledInSession(
        @PathVariable userId: Long,
        @PathVariable sessionId: Long
    ): Boolean?
}
